SECURITY.md

```diff
@@ -110,6 +112,42 @@ time, we will work towards improved safe-by-default configuration, but due to ba
 compatibility and performance concerns, this will require following the breaking change deprecation
 policy.
 
+#### Core and extensions
+
+Anything in the Envoy core may be used in both untrusted and trusted deployments. As a consequence,
+it should be hardened with this model in mind.  Security issues related to core code will usually
+trigger the security release process as described in this document.
+
+Some [extensions](EXTENSION_POLICY.md) are also widely used and deserve the same treatment as core.
+Examples include HTTP connection manager, TCP proxy, buffer filter, router filter, CORS, dynamic
+forward proxy, anything intended to enforce access control policies (RBAC, external authorization)
+and all transport socket extensions.
+
+Other extensions are only expected to be used in trusted environments today.  These include, but are
+not limited to most non-HTTP network filters (e.g.  Thrift, Dynamo, Mongo, Redis, Kafka, Dubbo,
+Zookeeper, MySQL), HTTP inspector, Lua and Squash. We will not treat any issues in these extensions
```

Comment:
FWIW I think we need to support Lua, a lot of people use it in prod in edge serving situations.

Reply:
What threat model? Do we support untrusted Lua scripts, or are we defensive only against data plane traffic to otherwise trusted Lua scripts? The same is likely to apply to WASM.

Reply:
Good clarification. I would say only trusted Lua scripts, but processing downstream untrusted data.

Reply:
+1 (at least for the time being)

Reply:
Yeah, that would match my intuition. Updated.